mentat/code_context.py

```diff
@@ -34,6 +39,53 @@ class CodeContextSettings:
     auto_tokens: Optional[int] = None
 
 
+def _get_all_features(
+    git_root: Path,
+    include_files: dict[Path, CodeFeature],
+    diff_context: DiffContext,
+    code_map: bool,
+    level: CodeMessageLevel,
+) -> list[CodeFeature]:
+    """Return a list of all features in the git root with given properties."""
+
+    _features = list[CodeFeature]()
+    for path in get_non_gitignored_files(git_root):
+        abs_path = git_root / path
+        if (
+            abs_path in include_files
+            or abs_path.is_dir()
+            or not is_file_text_encoded(abs_path)
+        ):
+            continue
+
+        diff_target = diff_context.target if abs_path in diff_context.files else None
+        user_included = path in include_files
```

Comment:
Won't this always be false? (first of all because of line 55 and second of all because include_files has absolute paths as keys)?

Reply:
You're right, good catch. I've added a test for this too.